packages-ui/roosterjs-react/lib/ribbon/plugin/createRibbonPlugin.ts

```diff
@@ -49,6 +50,21 @@ class RibbonPluginImpl implements RibbonPlugin {
      * @param event PluginEvent object
      */
     onPluginEvent(event: PluginEvent) {
+        if (
+            event.eventType == PluginEventType.KeyDown &&
+            event.rawEvent.key == 'k' &&
+            (event.rawEvent.ctrlKey || event.rawEvent.metaKey) &&
+            !event.rawEvent.altKey
+        ) {
+            if (insertLink.disableInsertLink) {
+                event.rawEvent.preventDefault();
+                return;
+            }
+
+            this.onButtonClick(insertLink, 'insertLinkTitle', undefined);
```

Comment:
Instead of directly invoking the button, you can extract the onButtonClick function of insertLink button to be a common function, and let both your code and the insertLinkButton to call that function. So RibbonPlugin doesn't need to have dependency to the button.